jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/jmx/WebAppProviderMBean.java

```diff
@@ -0,0 +1,44 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.deploy.providers.jmx;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.deploy.providers.WebAppProvider;
+import org.eclipse.jetty.server.handler.jmx.AbstractHandlerMBean;
+import org.eclipse.jetty.util.annotation.ManagedAttribute;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+
+@ManagedObject("WebAppProvider mbean wrapper")
+public class WebAppProviderMBean extends AbstractHandlerMBean
+{
+    public WebAppProviderMBean(Object managedObject)
+    {
+        super(managedObject);
+    }
+
+    @ManagedAttribute("List of monitored resource directories")
+    public List<String> getMonitoredDirectories()
+    {
+        return ((WebAppProvider) _managed).getMonitoredResources().stream()
+                .map((r) -> r.getURI().toASCIIString())
+                .collect(Collectors.toList());
```

Comment:
We can monitor also file resources, right? If so, then it's not correct to call this method `getMonitoredDirectories()`.

Reply:
Yup. good point. fixing it.